backend/Services/SightingsService.cs

```diff
@@ -119,16 +122,35 @@ public async Task ApproveSighting(int sightingId)
     {
         Sighting sighting = await GetSightingById(sightingId);
 
-        sighting.IsApproved = true;
-
-        try
+        if (!sighting.IsApproved)
         {
-            _context.Sightings.Update(sighting);
-            _context.SaveChanges();
+            sighting.IsApproved = true;
+
+            try
+            {
+                _context.Sightings.Update(sighting);
+                _context.SaveChanges();
+            }
+            catch
+            {
+                throw new InvalidOperationException($"Sighting with ID {sightingId} cannot be approved");
+            }
+
+            try
+            {
+                await _userservice.AddPoint(sighting.UserId.ToString());
+            }
```

Comment:
Awesome work! Our only question is if we need to add a `_context.SaveChanges();` after line 141, so that the changes brought by the AddPoint method are saved to the database?

Reply:
We don't feel that this was required since it was already part of addpoint (line: await _userManager.UpdateAsync(user); )